File: Icms.Domain/Entities/Department.cs
using Icms.Domain.Enums;

namespace Icms.Domain.Entities;

public class Department
{
    public long Id { get; set; }
    public long? ChurchId { get; set; }
    public required string Name { get; set; }
    public DepartmentType Type { get; set; } = DepartmentType.Spiritual;
    public long? HeadEmployeeId { get; set; }
    public bool IsDeleted { get; set; }

    public Church? Church { get; set; }
    public Employee? HeadEmployee { get; set; }
    public ICollection<DepartmentEmployee> Employees { get; set; } = new List<DepartmentEmployee>();
    public ICollection<DepartmentActivity> Activities { get; set; } = new List<DepartmentActivity>();
}

File: icms-web/src/app/models/department.ts
export type DepartmentType =
  | 'Spiritual'
  | 'Charity'
  | 'Development'
  | 'Administrative';

export type DepartmentEmployeeRole = 'Head' | 'Staff' | 'Secretary';

/** Mirrors DepartmentResponseDto. */
export interface Department {
  id: number;
  name: string;
  type: DepartmentType;
  headEmployeeId: number | null;
  headEmployeeName: string | null;
  employeeCount: number;
}

/** Body for POST/PUT department — mirrors CreateDepartmentRequest / UpdateDepartmentRequest. */
export interface UpsertDepartmentRequest {
  name?: string | null;
  type?: DepartmentType | null;
  headEmployeeId?: number | null;
}

/** Mirrors DepartmentEmployeeDto (assigned employee on a department). */
export interface DepartmentEmployee {
  id: number;
  employeeId: number;
  employeeName: string | null;
  employeePosition: string;
  role: DepartmentEmployeeRole;
}

/** Body for POST department/{id}/employees — mirrors AddDepartmentEmployeeRequest. */
export interface AddDepartmentEmployeeRequest {
  employeeId?: number | null;
  role?: DepartmentEmployeeRole | null;
}
